synapse/lib/storm.py

```diff
@@ -3165,26 +3171,101 @@ async def execStormCmd(self, runt, genr):
         async with contextlib.AsyncExitStack() as stack:
 
             runts = []
-            for text in self.opts.query:
+            query_arguments = await s_stormtypes.toprim(self.opts.query)
+            queries = []
+            for arg in query_arguments:
+                if isinstance(arg, str):
+                    queries.append(arg)
+                    continue
+                # if a argument is a container/iterable, we'll add
+                # whatever content is in it as query text
+                for text in arg:
+                    queries.append(text)
+
+            for text in queries:
                 query = await runt.getStormQuery(text)
                 subr = await stack.enter_async_context(runt.getSubRuntime(query))
                 runts.append(subr)
 
+            size = len(runts)
+            outq_size = size * 2
             item = None
-            async for item in genr:
+            async for node, path in genr:
+
+                if self.opts.parallel and runts:
+
+                    outq = asyncio.Queue(maxsize=outq_size)
+                    for subr in runts:
+                        subg = s_common.agen((node, path.fork(node)))
```

Comment:
this causes a schedCoro() per inbound node.  we should optimize for creating fewer runtimes and having 1 genr per subquery